### DISCUSSION TITLE
Python install?

### DISCUSSION THREAD

**Discussion by kacper0000** [smartcontractkit/full-blockchain-solidity-course-py#1681] [General]
I install Python from Link provided.
I typed pip --version in bash like Patryk.
bash: pip: command not found
Also after I get pip install black: 
bash: pip: command not found

Why it is so?

**Comment by Rodrigo-Henrique21** [comment]
@kacper0000  esse problema pode ter varias causas vou começar supondo que seu não pip está instalado;

Primeiro verifique se seu pip está na pasta;

navegue até a pasta de instalação do seu python e dentro dela acessar a pasta verifiquei sua pasta scripts
Precisa constar nela o pip.exe

Caso não conste o diagnostico é a falta da instalação do pip o troubleshooting  será instala-lo 

Agora caso esteja lá;

Vamos verificar a variável de ambiente , caso não esteja criada vamos cria-la sendo ela;  No nosso caso, é a pasta Scripts que está dentro da pasta de instalação do Python.

C:\CAMINHO\Scripts 

Uma outra situação que pode acontecer é o pip não estar instalado no seu windows. Ou seja, você não encontrou o executável do pip naquela pasta de Scripts que fica dentro da pasta de instalação do python.

Neste caso você precisa instalar o pip

1 - Baixe o script get-pip.py
2 - Abra o prompt de comandos(como adm) e navegue até a pasta onde você salvou o get-pip.py
3 - Execute    --  python get-pip.py
*Pronto, após esse comando o pip já foi instalado e adicionado na sua variável de ambiente PATH ( ou seja voce este caso precisa da variável PATH )
![Sem título](https://user-images.githubusercontent.com/95323435/187496211-24e5d9db-7020-4ce3-9503-dda03f303e77.png)


OBS: Variável de ambiente você pode achar nas propriedades avançadas segue imagem 

Refs:

https://dicasdepython.com.br/resolvido-pip-nao-e-reconhecido-como-um-comando-interno/#:~:text=Uma%20situa%C3%A7%C3%A3o%20que%20pode%20acontecer,gente%20usa%20o%20pr%C3%B3prio%20python.

https://jobu.com.br/2021/05/25/pip-nao-e-reconhecido/

https://pt.stackoverflow.com/questions/238850/pip-n%C3%A3o-%C3%A9-reconhecido-como-um-comando-interno-ou-externo-um-programa-oper%C3%A1vel

**Comment by cromewar** [comment]
Hello @kacper0000 
Well @Rodrigo-Henrique21 gave us a detailed explanation, but in Portuguese (the auto translate works just fine tough).
However, I think there is a simpler solution for you problem, it seems you are using bash which means you are most likely using Linux or WSL, may be MAC  but not probably as Mac uses zsh.
So please be sure you have installed correctly python and pip:
`sudo apt install python3 python3-pip ipython3`
Then in some cases when running pip you should use pip3 instead of pip, so you can try using:
`pip3 install install black`

If none of these works, you also should add the following command to your `.bashrc` file:

`export PATH=$HOME/usr/local/bin:$PATH`

Hope this info might help.

Cheers.
